client/overview/survey/index.tsx

```diff
@@ -0,0 +1,185 @@
+/** @format **/
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies.
+ */
+import {
+	Card,
+	CardBody,
+	Button,
+	TextareaControl,
+	Icon,
+} from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import './style.scss';
+import Emoticons from 'wcpay/overview/survey/emoticons';
+import { HorizontalRule } from '@wordpress/primitives';
+import { useOverviewSurveyContext } from './context';
+import { OverviewSurveyFields } from 'wcpay/overview/survey/types';
+import close from 'wcpay/overview/survey/icons/close';
+
+const Survey = () => {
+	const { status } = useOverviewSurveyContext();
+	const {
+		setSurveySubmitted,
+		surveySubmitted,
+		surveyAnswers,
+		setSurveyAnswers,
+	} = useOverviewSurveyContext();
+
+	const setReviewRating = function ( value: number ) {
+		const answers: OverviewSurveyFields = {
+			...surveyAnswers,
+			rating: value,
+		};
+		setSurveyAnswers( answers );
+		if ( value > 3 ) {
+			setSurveySubmitted( answers );
+		}
+	};
+
+	return (
+		<Card>
+			<CardBody className="wcpay-overview__survey__container">
+				{ ! surveySubmitted && (
+					<>
+						<div className="survey_container">
+							<div className="emoticons_container">
+								<span>
+									{ __(
+										'How do you like your new finance overview?',
+										'woocommerce-payments'
+									) }
+								</span>
+								<Emoticons
+									disabled={ 'pending' === status }
+									rating="1"
+									setReviewRating={ setReviewRating }
+									currentRating={ surveyAnswers.rating ?? 0 }
```

Comment:
Nitpick: I wonder if we declare something like `let currentRating = surveyAnswers.rating ?? 0` at the top of the `Survey` class (and update it if needed in `setReviewRating` to avoid needing to have this nullish coalescence operator in multiple places throught this file? 🙂

Reply:
changed in fixed in https://github.com/Automattic/woocommerce-payments/pull/8309/commits/2f32d7cf7f9d001384c5c61315e2c249ac863406 thank you!